sympy/stats/crv.py

```diff
@@ -444,12 +444,17 @@ def probability(self, condition, **kwargs):
         except NotImplementedError:
             from sympy.stats.rv import density
             expr = condition.lhs - condition.rhs
-            dens = density(expr, **kwargs)
+            comp = 0
+            if len(random_symbols(expr)) == 0:
+                dens = self.density
+                comp = condition.rhs
+            else:
+                dens = density(expr, **kwargs)
             if not isinstance(dens, ContinuousDistribution):
-                dens = ContinuousDistributionHandmade(dens)
+                dens = ContinuousDistributionHandmade(dens, set=self.domain.set)
```

Comment:
IMO, we should check for other types(discrete, continuous) of distributions for this kind of fix. Can you look into it?

Reply:
Yes, I will add test of few of other distributions too.